packages/evm/test/opcodes/profiler.ts

```diff
@@ -0,0 +1,103 @@
+/* eslint no-console: 0 */
+
+import { Chain, Common, Hardfork } from '@ethereumjs/common'
+import { hexToBytes } from '@ethereumjs/util'
+
+import { EVM } from '../../src/index.js'
+
+import { opcodeTests } from './tests.js'
+import {
+  createBytecode,
+  createOpcodeTest,
+  getOpcodeByte,
+  getOpcodeTestName,
+  makeLoopCode,
+} from './utils.js'
+
+// Config
+const hardfork = Hardfork.Shanghai
+const GAS_LIMIT = BigInt(30_000_000) // 30M gas (mainnet max gas per block)
+// End config
+
+const GAS_TARGET = 30_000_000
+const SECONDS = 12
+
+const MGasPerSecondTarget = GAS_TARGET / SECONDS / 1e6
+
+const common = new Common({ chain: Chain.Mainnet, hardfork })
+
+let testCount = 0
+for (const testSet in opcodeTests) {
+  const testCases = opcodeTests[testSet]
+  for (const opcodeName in testCases) {
+    const tests = testCases[opcodeName]
+    for (const _ of tests) {
+      testCount++
+    }
+  }
+}
+
+const POP = getOpcodeByte('POP')
+
+const profilerResults: {
+  testName: string
+  opcode: string
+  mGasPerSecond: number
+}[] = []
+
+let progress = 0
+
+async function profile() {
+  for (const testSet in opcodeTests) {
+    const testCases = opcodeTests[testSet]
+    for (const opcodeName in testCases) {
+      const tests = testCases[opcodeName]
+      for (const test of tests) {
+        const testName = getOpcodeTestName(opcodeName, test.stack, test.name)
+        const code = makeLoopCode(
+          createBytecode([createOpcodeTest(test.stack, opcodeName, 'none'), POP])
+        )
+        const evm = new EVM({ common, profiler: { enabled: true } })
```

Comment:
```suggestion
        common.events.removeAllListeners()
```

Since we're reusing the same `common` each time, we should clear listeners because the `evm` adds a new listener [here](https://github.com/ethereumjs/ethereumjs-monorepo/blob/15f7e1770cf3ea92ccee225b67d23cb28e96ff70/packages/evm/src/evm.ts#L190) and I see memory leak errors when I run the profiler script.